packages/lesswrong/lib/collections/dialogueChecks/schema.ts

```diff
@@ -35,7 +36,29 @@ const schema: SchemaType<DbDialogueCheck> = {
     nullable: false,
     canRead: ['members'],
     // Defined in server/resolvers/dialogueChecksResolvers.ts
-  }
+  },
+  matchPreference: resolverOnlyField({
+    type: 'DialogueMatchPreference',
+    graphQLtype: 'DialogueMatchPreference',
+    canRead: ['members', 'admins'],
+    resolver: async (dialogueCheck: DbDialogueCheck, args: void, context: ResolverContext) => {
+      const { DialogueMatchPreferences, DialogueChecks } = context;
+      const matchPreference = await DialogueMatchPreferences.findOne({dialogueCheckId: dialogueCheck._id});
+      return await accessFilterSingle(context.currentUser, DialogueMatchPreferences, matchPreference, context);
+    }
+  }),
+  matchingMatchPreference: resolverOnlyField({
```

Comment:
this field name is pretty confusing - I'd call it something like `reciprocalMatchPreference` or `matchedUserMatchPreference`